Mbox-Store.md

```diff
@@ -62,7 +62,13 @@ your PATH and the JDK is in /usr/java:
     cd mbox
     mvn
     cd native
-    mvn
+    mvn -P1.7
```

Comment:
1.7 profile should not be required

Reply:
Currently  it is necessary. There is one validation that forces to use JDK11 if I don't specify the profile:
```
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-version) @ libmbox ---
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
Detected JDK Version: 1.8.0-231 is not in the allowed range [11,).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.899 s
[INFO] Finished at: 2021-03-26T12:07:26+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-version) on project libmbox: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
```

I will need to modify a pom.xml file to remove this requirement.

Reply:
jdk 11 for build is required

Reply:
in the other words, the build is supposed to work also on JDK 11

Reply:
I have updated the doc. The JDK 11 fix is in this other PR:
https://github.com/eclipse-ee4j/mail/pull/541